src/core/exec-invoke.c

```diff
@@ -3600,11 +3600,11 @@ static int acquire_home(const ExecContext *c, uid_t uid, const char** home, char
         if (!c->working_directory_home)
                 return 0;
 
-        r = get_home_dir(buf);
+        r = get_home_dir(ret_buf);
```

Comment:
At this stage, the uid is not applied yet, so get_home_dir() provides the home directory of the root, no?
Though, this is not your fault.

Reply:
If `User=` is explicitly specified, the home returned by `get_fixed_user` will be used. I'll add a comment.

Reply:
Ah, indeed.

BTW, `get_fixed_user()` is (of course) called when DynamicUser= is disabled.
Maybe, we should refuse `DynamicUser=` with `WorkindDirectory=~` ?

Reply:
Hmm, is `DynamicUser=` supposed to work with unprivileged users? It seems that the manager doesn't refuse the unit but sd-executor fails half-way, when trying to take a lock on `/etc/.pwd.lock`?